sqlglot/generator.py

```diff
@@ -4005,3 +4005,6 @@ def parsejson_sql(self, expression: exp.ParseJSON) -> str:
             return self.sql(expression.this)
 
         return self.func(self.PARSE_JSON_NAME, expression.this, expression.expression)
+
+    def length_sql(self, expression: exp.Length) -> str:
+        return self.func("LENGTH", expression.this)
```

Comment:
why does the default skip binary?

Reply:
is binary a non renderable arg?

Reply:
`binary` is a boolean flag that's added from BigQuery and Snowflake because they also accept `LENGTH(<binary_expr>)` alongside `LENGTH(<string_expr>)`, it's not a valid argument. 

Dialects like Redshift and DuckDB only accept strings, so if `binary` is set DDB will attempt to differentiate between the 2 by returning `LENGTH(<string_expr>)` or `OCTET_LENGTH(<binary_expr>)`

Reply:
we're also gonna crash without this because it's a boolean, see https://github.com/tobymao/sqlglot/blob/main/sqlglot/generator.py#L798